### DISCUSSION TITLE
How to reuse message when amending the last commit?

### DISCUSSION THREAD

**Discussion by salehi-hassan** [jesseduffield/lazygit#3671] [Q&A] [answered]
I am wondering what's the quickest way to undo the last commit, apply changes and commit agin with the old commit message.

So far, the solution I found is I can go to the 'Reflog' and copy the deleted commit message. Is there any quicker approach?

In VSCode, when you undo a commit using its built-in source control tool, it always keep the message in the commit section for the next commit.

**Comment by mark2185** [comment, marked as answer]
If you just want to add/modify changes, press `A` in the `Files` panel.

For removing things from the commit, take a look at the [tutorial](https://youtu.be/CPLdltN7wgE) feature called "Custom patch".

**Comment by stefanhaller** [comment]
A few more options, none of which are really great:

- Instead of dropping the commit, you could revert it; then, when making the new commit, press up-arrow twice to recall its commit message. After making the commit, range-select the original commit and the revert, and hit `d` to drop both at once.
- Instead of taking the commit message from the reflog, you could copy it from the old commit _before_ dropping it. However, pasting the message into the new commit is still cumbersome either way; I have opened #3672 to address this.
- Instead of dropping the old commit, you could make it empty, so that you can amend your new changes into it using shift-A. To make the commit empty, press enter on it, then shift-down to select all files, and `d` to drop them all.

**Comment by salehi-hassan** [comment]
Thanks all! The more I think the more I understand there was no need to reuse the message if I actually amend it 😄 The habit comes from the fact, I used to run 'git reset --soft' and then stage the changes and commit again.
